src/charm.py

```diff
@@ -83,7 +88,7 @@ def _config(self) -> str:
         db_info = self._get_database_relation_info()
 
         config = {
-            "dsn": f"postgres://{db_info['username']}:{db_info['password']}@{db_info['endpoints']}/postgres",
+            "dsn": f"postgres://{db_info['username']}:{db_info['password']}@{db_info['endpoints']}/{self._name}",
```

Comment:
I am not sure if there will ever be a problem with this, but I think that `{self.model.name}_{self._name}` would be safer here (to avoid collisions with apps of the same name running in different models.